frontend/src/cljs/imas_seamap/map/events.cljs

```diff
@@ -277,12 +277,25 @@
   (let [db (assoc-in db [:map :base-layers] layers)]
     (update-grouped-base-layers db)))
 
+(defn- keyed-layers-join
+  "Using layers and keyed-layers, replaces layer IDs in keyed-layers with layer
+   objects. We only update keyed-layers if the db contains layers."
```

Comment:
That comment made me think (because yes, race conditions have been a factor before).  There's no real reason to expand these into layers up-front, it just means the sub becomes "return this app-db path", but you could defer that to sub-time if you want.

Reply:
> I just had to triple-check that it's not covid-brain, but there's a lot of overlap between #105 and #106! (eg the django side of things)

Yeah, I added a comment on #106 where I said that this PR was a prerequisite because I needed some of the code functionality. I'm now noting that it's causing some issues when you're reviewing the file diffs of the pull requests because it's saying the code changes are in both. In future I'll just branch off these feature branches if I need features that haven't yet been merged into develop (unfortunately I already made PR #108, so you're going to see weird code diffs in that that look like #107).